cupy/_indexing/generate.py

```diff
@@ -70,10 +64,8 @@ def __getitem__(self, key):
                         newobj = self._output_obj(newobj, ndim, ndmin, trans1d)
 
             objs.append(newobj)
-            if not scalar and isinstance(newobj, _core.ndarray):
-                arraytypes.append(newobj.dtype)
 
-        final_dtype = numpy.find_common_type(arraytypes, scalartypes)
+        final_dtype = numpy.result_type(*key)
```

Comment:
Thanks, changes look good!  This one, I suspect this is slightly more complicated because `key` can include slice objects.  I am happy to also make a PR (was going to look into it later).  The change I did in NumPy is here:

https://github.com/numpy/numpy/pull/22539/files#diff-8bea6cb955d7704ddbe7810c30cb214e48574730c1304ee17bdf09e61be40f8d

Reply:
Thanks! We don't support slice objects here so it should be OK for now 😄 
https://github.com/cupy/cupy/blob/c5153de285c03ed4824c14da6c85b5d31040cb76/cupy/_indexing/generate.py#L49-L50